core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -171,11 +178,22 @@ func (ec *EthConfirmer) processHead(ctx context.Context, head models.Head) error
 	logger.Debugw("EthConfirmer: finished RebroadcastWhereNecessary", "headNum", head.Number, "time", time.Since(mark), "id", "eth_confirmer")
 	mark = time.Now()
 
-	defer func() {
-		logger.Debugw("EthConfirmer: finished EnsureConfirmedTransactionsInLongestChain", "headNum", head.Number, "time", time.Since(mark), "id", "eth_confirmer")
-	}()
+	if err := errors.Wrap(ec.EnsureConfirmedTransactionsInLongestChain(ctx, head), "EnsureConfirmedTransactionsInLongestChain failed"); err != nil {
+		return err
+	}
+
+	logger.Debugw("EthConfirmer: finished EnsureConfirmedTransactionsInLongestChain", "headNum", head.Number, "time", time.Since(mark), "id", "eth_confirmer")
+
+	if ec.resumeCallback != nil {
```

Comment:
I wonder if this could be async? Probably won't take too long so it doesn't matter for now

Reply:
Yeah I'd prefer to keep this as is for now. This way all the runs are guaranteed to be running by the time that the confirmer loop completes (and won't be picked up in a subsequent iteration)

Reply:
hm `ec.resumeCallback` is supposed to be always not nil here, right? You could add an assert in NewEthConfirmer